modules/nf-core/doubletdetection/main.nf

```diff
@@ -0,0 +1,49 @@
+process DOUBLETDETECTION {
+    tag "$meta.id"
+    label 'process_medium'
+
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'oras://community.wave.seqera.io/library/anndata_louvain_pip_doubletdetection:42d2326cc250350b':
+        'community.wave.seqera.io/library/anndata_louvain_pip_doubletdetection:cbe92394c10372fa' }"
+
+    // Prevent /tmp mount for singularity and set the MATPLOTLIB TMPDIR
+    containerOptions "${workflow.containerEngine == 'singularity' ?
```

Comment:
@nictru we talked about the `containerOptions` and how to use them in this example and also other PRs (https://github.com/nf-core/modules/pull/6891)
I basically see the following options:

1. Move the definition of the variables inside the python script, s.t. it gets executed regardless of the executer. Then we should still keep the `--no-mount tmp` option (at least for singularity and apptainer) unless we are really sure nothing writes into the mounted `tmp` directory to avoid conflicts with other users using the same process at the same time. This could be done either on pipeline level or on process level (preferred).
2. Set the environment variables in the process itself (as implemented right now), but regardless of which container type is used (remove `if`-block). Docker, Singularity and Apptainer definitely support the `--env` flag, however this is not clear for all other container types (Podman,, Sarus, ...). The `--no-mount tmp` argument would probably be defined on pipeline level, as not all container types support it.
3. Leave it as it is implemented right now. This is not optimal, as apptainer and docker will have their temporary data stored at different locations then singularity.

Reply:
Let's see what insights arise from [this thread](https://nfcore.slack.com/archives/CE6SDBX2A/p1730720790953729)